core/trino-main/src/main/java/io/trino/sql/planner/RelationPlanner.java

```diff
@@ -678,38 +687,44 @@ public PatternRecognitionComponents planPatternRecognitionComponents(
+        for (SubsetDefinition subsetDefinition : subsets) {
+            String label = analysis.getResolvedLabel(subsetDefinition.getName());
+            Set<IrLabel> elements = analysis.getLabels(subsetDefinition).stream()
+                    .map(IrLabel::new)
+                    .collect(toImmutableSet());
+            rewrittenSubsetsBuilder.put(new IrLabel(label), elements);
+        }
+        Map<IrLabel, Set<IrLabel>> rewrittenSubsets = rewrittenSubsetsBuilder.buildOrThrow();
+
         // rewrite measures
         ImmutableMap.Builder<Symbol, Measure> rewrittenMeasures = ImmutableMap.builder();
         ImmutableList.Builder<Symbol> measureOutputs = ImmutableList.builder();
-        ImmutableMap<IrLabel, Set<IrLabel>> rewrittenSubsets = rewrittenSubsetsBuilder.buildOrThrow();
-        for (MeasureDefinition measureDefinition : measures) {
-            Type type = analysis.getType(measureDefinition.getExpression());
-            Symbol symbol = symbolAllocator.newSymbol(measureDefinition.getName().getValue().toLowerCase(ENGLISH), type);
-            Expression expression = expressionRewrite.apply(measureDefinition.getExpression());
-            ExpressionAndValuePointers measure = LogicalIndexExtractor.rewrite(expression, rewrittenSubsets, symbolAllocator, plannerContext.getMetadata());
+
+        for (MeasureDefinition definition : measures) {
+            Type type = analysis.getType(definition.getExpression());
+            Symbol symbol = symbolAllocator.newSymbol(definition.getName().getValue().toLowerCase(ENGLISH), type);
+            ExpressionAndValuePointers measure = planPatternRecognitionExpression(translations, rewrittenSubsets, definition.getName().getValue().toLowerCase(ENGLISH), definition.getExpression());
             rewrittenMeasures.put(symbol, new Measure(measure, type));
             measureOutputs.add(symbol);
         }
 
-        // rewrite pattern to IR
-        IrRowPattern rewrittenPattern = RowPatternToIrRewriter.rewrite(pattern, analysis);
-
         // rewrite variable definitions
         ImmutableMap.Builder<IrLabel, ExpressionAndValuePointers> rewrittenVariableDefinitions = ImmutableMap.builder();
-        for (VariableDefinition variableDefinition : variableDefinitions) {
-            IrLabel label = irLabel(variableDefinition.getName());
-            Expression expression = expressionRewrite.apply(variableDefinition.getExpression());
-            ExpressionAndValuePointers definition = LogicalIndexExtractor.rewrite(expression, rewrittenSubsets, symbolAllocator, plannerContext.getMetadata());
-            rewrittenVariableDefinitions.put(label, definition);
+        for (VariableDefinition definition : variableDefinitions) {
+            String label = analysis.getResolvedLabel(definition.getName());
+            ExpressionAndValuePointers variable = planPatternRecognitionExpression(translations, rewrittenSubsets, definition.getName().getValue().toLowerCase(ENGLISH), definition.getExpression());
```

Comment:
Prefer `ROOT` to `ENGLISH`

Reply:
I've removed them. The toLowerCase() calls are actually not needed.